sydent/threepid/bind.py

```diff
@@ -101,29 +102,27 @@ def _notify(self, assoc, attempt):
         mxid = assoc["mxid"]
         domain = mxid.split(":")[-1]
         server = yield self._pickServer(domain)
-        callbackUrl = "https://%s/_matrix/federation/v1/3pid/onbind" % (
+
+        post_url = "https://%s/_matrix/federation/v1/3pid/onbind" % (
             server,
         )
 
         logger.info("Making bind callback to: %s", callbackUrl)
-        # TODO: Not be woefully insecure
-        agent = Agent(reactor, InsecureInterceptableContextFactory())
-        reqDeferred = agent.request(
-            "POST",
-            callbackUrl.encode("utf8"),
-            Headers({
+
+        # Make a POST to the chosen Synapse server
+        http_client = FederationHttpClient()
+        response = yield http_client.post_json_get_nothing(post_url, assoc, {
+            'headers': {
                 "Content-Type": ["application/json"],
                 "User-Agent": ["Sydent"],
-            }),
-            FileBodyProducer(StringIO(json.dumps(assoc)))
-        )
-        reqDeferred.addCallback(
-            lambda _: logger.info("Successfully notified on bind for %s" % (mxid,))
-        )
+            }
+        })
 
-        reqDeferred.addErrback(
-            lambda err: self._notifyErrback(assoc, attempt, err)
-        )
+        # If the request failed, try again with exponential backoff
+        if response.code != 200:
```

Comment:
do we not also need to do this for requests which do not result in a valid HTTP response at all (eg, connection errors)?

Reply:
Handled by above try/catch.